sdk/export/metric/aggregation/temporality.go

```diff
@@ -0,0 +1,119 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:generate stringer -type=Temporality
+
+package aggregation // import "go.opentelemetry.io/otel/sdk/export/metric/aggregation"
+
+import (
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/sdkapi"
+)
+
+// Temporality indicates the kind of data exported by an exporter.
+// These bits may be OR-d together when multiple exporters are in use.
+type Temporality int
```

Comment:
Should this be an integer with a defined length so it is not represented differently on different machines? Also, if we make this unsigned it would allow for an additional bit of data.

```suggestion
type Temporality uint8
```

Reply:
I would agree not because of the size, but because of the culture of using `uints` for this.  Generally, bitfields aren't ints.

Reply:
Done.